core/src/main/java/org/mobilitydata/gtfsvalidator/model/ValidationReport.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.model;
+
+import com.google.common.collect.Sets;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.mobilitydata.gtfsvalidator.io.ValidationReportDeserializer;
+
+/**
+ * Used to (de)serialize a {@code NoticeContainer}. This represents a validation report as a list of
+ * {@code NoticeReport} which provides information about each notice generated during a GTFS dataset
```

Comment:
> nit: I'd avoid putting too much information on what NoticeReport is since this will require to keep this line updated, which less likely to happen.

👍🏾

> In the next line what "This objects stores" is referring to?

This `ValidationReport`, the docstrings have been updated for clarification.